### ISSUE TITLE
Miniconda installer cannot install Start Menu items on windows

### ISSUE THREAD

**Issue by mcg1969** [ContinuumIO/anaconda-issues#136]
First posted as https://github.com/conda/conda/issues/782; thank you to @asmeurer for directing me here instead.

Miniconda does not install `menuinst`; and as a result, it does not install Start Menu items _even though it says it will_. And because of the known filelock issue, menuinst cannot be installed after the fact.

Therefore, it seems that one of two things should happen (my vote is for the first):
- `menuinst` should be one of the packages included with miniconda.
- The installer text that claims that a Start Menu item will be created should be removed.

**Comment by cmaes** [comment]
I ran into this issue as well. I installed Miniconda and was surprised to see it did not install a shortcut to a shell in the Start menu (even though the installer said it would).  I'd like to request that the confusing text in the installer be removed, or menuinst be included so Miniconda could add something to the Start Menu.

**Comment by msarahan** [comment]
Pending fix:

conda/conda#1699
ContinuumIO/menuinst#4
conda/conda-recipes#439
https://github.com/conda/conda-env/pull/178

**Comment by H0R5E** [comment]
```
d:\Usuarios\108630>conda info
Current conda install:

             platform : win-64
        conda version : 3.19.1
  conda-build version : 1.19.0
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : C:\Anaconda  (writable)
  default environment : C:\Anaconda
     envs directories : C:\Anaconda\envs
        package cache : C:\Anaconda\pkgs
         channel URLs : https://conda.binstar.org/t/<TOKEN>/topper/win-64/
                        https://conda.binstar.org/t/<TOKEN>/topper/noarch/
                        https://repo.continuum.io/pkgs/free/win-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : d:\Usuarios\108630\.condarc
    is foreign system : False
```

So, the situation is now that when upgrading packages that need permissions to install (update?) windows menus it will request an increase in permissions with a dialogue box. That's good. Unfortunately, there are some bad things too.

Mainly, I don't want these packages to try to install menu shortcuts outside of the root environment. Maybe some people want that, but I use so many environments that it would be nice to turn off that behaviour. I am also worried that its not clear which environment the menu shortcuts are associated to, if you have the same package in more than one (I guess the last to be installed, and that if you uninstall, the links to the other environment will not be rebuilt).

**Comment by msarahan** [comment]
Agreed, with lots of environments it kind of breaks down.  I think the next iteration of this is to create a folder for each environment's shortcuts inside the anaconda folder.

Shortcuts should have the environment name appended to the end of the shortcut name.  If the string is too long and Windows cuts it off, then we need a different way (probably the folder solution)

If you name your environments starting with an underscore ( _env_name ) then shortcuts will not get installed.  A more explicit flag would be better.  I'll add an issue for that.  https://github.com/conda/conda/issues/2089

**Comment by gansanay** [comment]
Hi,

This so-called feature is exactly what I would like to avoid. My builds and tests are distributed to gitlab-runner instances on windows hosts, and not only it messes with their Start Menus, which is unwanted, but it also fails:

```
INFO menuinst_win32:__init__(182): Menu: name: 'Anaconda${PY_VER} ${PLATFORM}', prefix: 'D:\Multi-Runner\builds\XXX', env_name: 'XXX', mode: 'None', used_mode: 'user'
INFO menuinst_win32:folder_path(73): WARNING: menuinst key: 'start'
                 path: 'C:\Windows\system32\config\systemprofile\AppData\Roaming\Microsoft\Windows\Start Menu\Programs'
     .. excepted with: 'PathNotVerifiableException' in knownfolders.py, implementing workarounds ..
INFO menuinst_win32:folder_path(98):      .. due to lack of 'check_other_mode' not picking
        non-excepting path of 'C:\ProgramData\Microsoft\Windows\Start Menu\Programs'
 in knownfolders.py
INFO menuinst_win32:folder_path(73): WARNING: menuinst key: 'quicklaunch'
                 path: 'C:\Windows\system32\config\systemprofile\AppData\Roaming\Microsoft\Internet Explorer\Quick Launch'
     .. excepted with: 'PathNotVerifiableException' in knownfolders.py, implementing workarounds ..
menuinst ExceptionINFO menuinst_win32:__init__(182): Menu: name: 'Anaconda${PY_VER} ${PLATFORM}', prefix: 'D:\Multi-Runner\builds\XXX', env_name: 'XXX', mode: 'None', used_mode: 'user'
INFO menuinst_win32:folder_path(73): WARNING: menuinst key: 'start'
                 path: 'C:\Windows\system32\config\systemprofile\AppData\Roaming\Microsoft\Windows\Start Menu\Programs'
     .. excepted with: 'PathNotVerifiableException' in knownfolders.py, implementing workarounds ..
INFO menuinst_win32:folder_path(98):      .. due to lack of 'check_other_mode' not picking
        non-excepting path of 'C:\ProgramData\Microsoft\Windows\Start Menu\Programs'
 in knownfolders.py
INFO menuinst_win32:folder_path(73): WARNING: menuinst key: 'quicklaunch'
                 path: 'C:\Windows\system32\config\systemprofile\AppData\Roaming\Microsoft\Internet Explorer\Quick Launch'
     .. excepted with: 'PathNotVerifiableException' in knownfolders.py, implementing workarounds ..
```

My main concern is at the moment: how do I disable it?

**Comment by stevengj** [comment]
This is also a problem for us in Julia, where we use miniconda to install an "internal" miniconda distro in order to supply dependencies, and our internal miniconda installation should not be user visible in the Start menu: https://github.com/JuliaPy/Conda.jl/issues/147

Has anyone figured out a way to prevent the miniconda installer from touching the Start menu?

**Comment by msarahan** [comment]
If your environment name starts with an underscore, no shortcuts are installed.  There's also a flag for conda, but that doesn't help with miniconda. On my phone, so I can't look it up easily, but look through the output of 


Conda install -h

And

Conda create -h

If you need a flag for the installer itself, please consider submitting a PR to GitHub.com/conda/constructor

**Comment by stevengj** [comment]
Beginning an environment name with an underscore doesn't seem applicable to installing the root environment?

A flag for the installer itself would be ideal.
